### DISCUSSION TITLE
Get mask of what is behind a mesh? (GrabPass)

### DISCUSSION THREAD

**Discussion by aVersionOfReality** [bnpr/Malt#475] [Q&A]
The Goo build of Eevee has a Screenspace info node that is capable of providing a mask of what is behind a mesh. In Unity, this is apparently called a [Grab Pass.](https://docs.unity3d.com/Manual/SL-GrabPass.html)

This can be used for all sorts of useful tricks. Here is an example of it being used to make fake cast shadows for hair based off a proxy mesh: https://twitter.com/festivizing/status/1617389991432458240

Does Malt have this? Or the option to create it? We do not necessarily need it for shadows, but it could be useful for all sorts of other things where we want to use geometry to define a mask.

**Comment by pragma37** [comment]
I'm not sure about what EEVEE Goo does exactly, are there docs anywhere?

That Unity Grab Pass looks a lot like this: https://github.com/bnpr/Malt/discussions/238#discussioncomment-2151256
Not sure if it's exactly the same, though.

**Reply by aVersionOfReality** [reply]
Unfortunately they don't seem to have any documentation. I'll see if I can find out more from them.
For my purposes, any method of creating a stencil/mask based on a mesh is fine.

There's a brief description here but its not technical documentation by any means:
https://youtu.be/PBdvcP8WhX8?t=1015

And the actual source files on github:
https://github.com/dillongoostudios/goo-engine
